### PR TITLE
make controller deployment run as nobody and nogroup

### PR DESCRIPTION
**Pull request by winston0410**
At the moment, the controller of this operator are running in nonRoot in deployment's securityContext, but the root filesystem is still readable, the user and group are not defined.

By setting 65334 uid and gid (nobody and nogroup), we make sure the controller will be running with least privilege. Also I set the fsGroup to 65334, so we make sure we turn all mounted volumes owned by nobody instead of root.

@jodevsa I am not sure if I have changed the manifest in the right place. If `/bundle` is generated, where should I edit instead? Also are you happy with making this operator tight in security by default? Or do you prefer leaving that for users to patch themselves?

### CODE REVIEW

**Comment by jodevsa** [comment]
Hi @winston0410,


it's been a long time since i last touched, went over the logic, but can you try to run `make bundle` and see if those changes are overwritten?

**Review by jodevsa — APPROVED** [review]
LGTM, please check if those aren't overwitten by `make bundle` before merging :) Thanks!

**Comment by winston0410** [comment]
> make bundle

Hey just checked the make file, seems like there is no command for `bundle`? What command should I run instead? Just want to make sure I don't make anything wrong before merging:

```sh
❯ make help

Usage:
  make <target>

General
  help             Display this help.

Development
  manifests        Generate WebhookConfiguration, ClusterRole and CustomResourceDefinition objects.
  generate         Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
  fmt              Run go fmt against code.
  vet              Run go vet against code.
  test             Run tests.

Build
  build-agent      Build manager binary.
  build-manager    Build manager binary.
  run              Run a controller from your host.
  docker-build-agent  Build docker image with the manager.
  docker-build-manager  Build docker image with the manager.
  docker-push      Push docker image with the manager.

Deployment
  install          Install CRDs into the K8s cluster specified in ~/.kube/config.
  uninstall        Uninstall CRDs from the K8s cluster specified in ~/.kube/config.
  deploy           Deploy controller to the K8s cluster specified in ~/.kube/config.
  undeploy         Undeploy controller from the K8s cluster specified in ~/.kube/config.
  controller-gen   Download controller-gen locally if necessary.
  kind             Download kind locally if necessary.
  kustomize        Download kustomize locally if necessary.
  envtest          Download envtest-setup locally if necessary.
```

**Comment by jodevsa** [comment]
Hi @winston0410, 

I just checked, we currently don't support the opereator-sdk bundle 🤔 



So, you are interested in making this change to the operator, like this pod:
```
wireguard-system   wireguard-controller-manager-648ff4d55d-lhtzg   2/2     Running   0          16h

```

??

I think the files for the manager deployment are in:  https://github.com/jodevsa/wireguard-operator/tree/main/config/manager

**Comment by jodevsa** [comment]
so, this one: https://github.com/jodevsa/wireguard-operator/blob/main/config/manager/manager.yaml

**Comment by jodevsa** [comment]
and in the makefile, we use `generate-release-file` to create the release which would contain the manager

**Comment by jodevsa** [comment]
the bundle files are obselete for now as the project still haven't implemented this: https://sdk.operatorframework.io/docs/cli/operator-sdk_generate_bundle/



The bundle feature is useful to streamline publishing the operator but somehow the whole thing is so complicated that I haven't gotten around it :D

**Review by jodevsa — APPROVED** [review]

**Comment by winston0410** [comment]
> generate-release-file

I see, no worries. I have now moved the changes to the corresponding source files. Do I have to run `make generate-release-file` myself and commit the changes for the `./release.yaml`? Seems to be handled by the pipeline here?
https://github.com/jodevsa/wireguard-operator/blob/4763c0afc55ba2089cd974fc6e550f1b8620ac5b/.github/workflows/manual-dev-release-workflow.yaml#L65

**Pull request merged**
